homeassistant/components/schluter/climate.py

```diff
@@ -0,0 +1,165 @@
+    def should_poll(self):
+        """Return if platform should poll."""
+        return False
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._support_flags
+
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return self.coordinator.data[self.idx].serial_number
+
+    @property
+    def device_info(self):
+        """Return information about the device."""
+        return {
+            "identifiers": {(DOMAIN, self.coordinator.data[self.idx].serial_number)},
+            "name": self.coordinator.data[self.idx].name,
+            "manufacturer": "Schluter",
+            "model": "Thermostat",
+            "sw_version": self.coordinator.data[self.idx].sw_version,
+        }
+
+    @property
+    def name(self):
+        """Return the name of the thermostat."""
+        return self.coordinator.data[self.idx].name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.coordinator.data[self.idx].temperature
+
+    @property
+    def hvac_mode(self):
+        """Return current mode. Only heat available for floor thermostat."""
+        return HVAC_MODE_HEAT
+
+    @property
+    def hvac_action(self):
+        """Return current operation. Can only be heating or idle."""
+        return (
+            CURRENT_HVAC_HEAT
+            if self.coordinator.data[self.idx].is_heating
+            else CURRENT_HVAC_IDLE
+        )
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self.coordinator.data[self.idx].set_point_temp
+
+    @property
+    def hvac_modes(self):
+        """List of available operation modes."""
+        return [HVAC_MODE_HEAT]
+
+    @property
+    def min_temp(self):
+        """Identify min_temp in Schluter API."""
+        return self.coordinator.data[self.idx].min_temp
+
+    @property
+    def max_temp(self):
+        """Identify max_temp in Schluter API."""
+        return self.coordinator.data[self.idx].max_temp
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Mode is always heating, so do nothing."""
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        target_temp = None
+        target_temp = kwargs.get(ATTR_TEMPERATURE)
+        serial_number = self.coordinator.data[self.idx].serial_number
+        _LOGGER.debug("Setting thermostat temperature: %s", target_temp)
+
+        try:
+            if target_temp is not None:
+                self._api.set_temperature(self._session_id, serial_number, target_temp)
+        except RequestException as ex:
+            _LOGGER.error("An error occurred while setting temperature: %s", ex)
+            self.schedule_update_ha_state(True)
```

Comment:
Since your entity doesn't poll, you can directly call the write state function. However, what will change that causes data in the coordinator to change ? 

```suggestion
            self.async_write_ha_state()
```

Reply:
Based on a coordinator tutorial I was using, I thought this was necessary. After some testing I think you are right, I don't think it is necessary. I've removed it